win32/tests/threadpool_win32_ut/threadpool_win32_ut.c

```diff
@@ -375,23 +393,28 @@ TEST_FUNCTION(threadpool_dispose_frees_resources)
     ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
 }
 
-/* Tests_SRS_THREADPOOL_WIN32_01_006: [ While threadpool is OPENING or CLOSING, threadpool_dispose shall wait for the open to complete either successfully or with error. ]*/
 /* Tests_SRS_THREADPOOL_WIN32_01_007: [ threadpool_dispose shall perform an implicit close if threadpool is OPEN. ]*/
+/* Tests_SRS_THREADPOOL_WIN32_01_030: [ threadpool_close shall wait for any executing callbacks by calling CloseThreadpoolCleanupGroupMembers, passing FALSE as fCancelPendingCallbacks. ]*/
+/* Tests_SRS_THREADPOOL_WIN32_01_032: [ threadpool_close shall close the threadpool cleanup group by calling CloseThreadpoolCleanupGroup. ]*/
+/* Tests_SRS_THREADPOOL_WIN32_01_033: [ threadpool_close shall destroy the thread pool environment created in threadpool_create. ]*/
 TEST_FUNCTION(threadpool_dispose_performs_an_implicit_close)
 {
     // arrange
-    THANDLE(THREADPOOL) threadpool = threadpool_create(test_execution_engine);
     PTP_CALLBACK_ENVIRON cbe;
     PTP_CLEANUP_GROUP test_cleanup_group;
-    umock_c_reset_all_calls();
-
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG));
+    STRICT_EXPECTED_CALL(interlocked_exchange(IGNORED_ARG, IGNORED_ARG));
+    STRICT_EXPECTED_CALL(execution_engine_win32_get_threadpool(test_execution_engine));
     STRICT_EXPECTED_CALL(mocked_InitializeThreadpoolEnvironment(IGNORED_ARG))
         .CaptureArgumentValue_pcbe(&cbe);
     STRICT_EXPECTED_CALL(mocked_SetThreadpoolCallbackPool(IGNORED_ARG, test_pool))
         .ValidateArgumentValue_pcbe(&cbe);
     STRICT_EXPECTED_CALL(mocked_CreateThreadpoolCleanupGroup())
         .CaptureReturn(&test_cleanup_group);
-    (void)threadpool_open(threadpool);
+    STRICT_EXPECTED_CALL(mocked_SetThreadpoolCallbackCleanupGroup(IGNORED_ARG, IGNORED_ARG, IGNORED_ARG))
+        .ValidateArgumentValue_ptpcg(&test_cleanup_group);
+    STRICT_EXPECTED_CALL(execution_engine_inc_ref(test_execution_engine));
+    THANDLE(THREADPOOL) threadpool = threadpool_create(test_execution_engine);

```

Comment:
>threadpool [](http://example.com/codeflow?start=24&length=10)

There should be an assert is not null on this.

Reply:
Done.